### PR TITLE
Add a Prometheus Operator Component

### PR DESCRIPTION
**Pull request by sdwilsh**

### CODE REVIEW

File: kustomization/components/operator-prometheus/subscription.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: my-prometheus
```
Comment (robarnold) [inline kustomization/components/operator-prometheus/subscription.yml:5]: Never did like the my prefix on names. Can we drop it?

File: kustomization/components/operator-prometheus/subscription.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  namespace: operators
+spec:
+  channel: beta
+  name: prometheus
+  source: operatorhubio-catalog
+  sourceNamespace: olm
```
Comment (robarnold) [inline kustomization/components/operator-prometheus/subscription.yml:11]: Not being familiar with Prometheus or the operator framework, these lines could use some commenting to explain what they do and how to customize them (if possible, necessary), or perhaps docs on the operator framework component itself.

**Review by robarnold — APPROVED** [review]

File: kustomization/components/operator-prometheus/subscription.yml (hunk 1 above)
Comment (sdwilsh) [reply kustomization/components/operator-prometheus/subscription.yml:5]: Sure; I guess this should really be something like `prometheus-subscription`

File: kustomization/components/operator-prometheus/subscription.yml (hunk 2 above)
Comment (sdwilsh) [reply kustomization/components/operator-prometheus/subscription.yml:11]: This is taken verbatim from the install instructions on https://operatorhub.io/operator/prometheus

The `operatorhubio-catalog` is part of the olm installation, which lives in the `olm` namespace.

**Comment by sdwilsh** [comment]
We are not using OLM now, so this goes bye-bye

**Pull request closed without merging**